lib/dgram.js

```diff
@@ -63,88 +63,37 @@ const errnoException = errors.errnoException;
 const exceptionWithHostPort = errors.exceptionWithHostPort;
 
 
-function lookup4(lookup, address, callback) {
-  return lookup(address || '127.0.0.1', 4, callback);
-}
-
-
-function lookup6(lookup, address, callback) {
-  return lookup(address || '::1', 6, callback);
-}
-
-
-function newHandle(type, lookup) {
-  if (lookup === undefined) {
-    if (dns === undefined) dns = require('dns');
-    lookup = dns.lookup;
-  } else if (typeof lookup !== 'function')
-    throw new ERR_INVALID_ARG_TYPE('lookup', 'Function', lookup);
-
-  if (type === 'udp4') {
-    const handle = new UDP();
-    handle.lookup = lookup4.bind(handle, lookup);
-    return handle;
-  }
-
-  if (type === 'udp6') {
-    const handle = new UDP();
-    handle.lookup = lookup6.bind(handle, lookup);
-    handle.bind = handle.bind6;
-    handle.send = handle.send6;
-    return handle;
-  }
-
-  throw new ERR_SOCKET_BAD_TYPE();
-}
-
-
-function _createSocketHandle(address, port, addressType, fd, flags) {
-  // Opening an existing fd is not supported for UDP handles.
-  assert(typeof fd !== 'number' || fd < 0);
-
-  var handle = newHandle(addressType);
-
-  if (port || address) {
-    var err = handle.bind(address, port || 0, flags);
-    if (err) {
-      handle.close();
-      return err;
-    }
-  }
-
-  return handle;
-}
-
-const kOptionSymbol = Symbol('options symbol');
-
 function Socket(type, listener) {
   EventEmitter.call(this);
+  const state = {};
   var lookup;
 
-  this[kOptionSymbol] = {};
   if (type !== null && typeof type === 'object') {
     var options = type;
     type = options.type;
     lookup = options.lookup;
-    this[kOptionSymbol].recvBufferSize = options.recvBufferSize;
-    this[kOptionSymbol].sendBufferSize = options.sendBufferSize;
+    state.recvBufferSize = options.recvBufferSize;
+    state.sendBufferSize = options.sendBufferSize;
   }
 
   var handle = newHandle(type, lookup);
   handle.owner = this;
 
-  this._handle = handle;
-  this._receiving = false;
-  this._bindState = BIND_STATE_UNBOUND;
-  this[async_id_symbol] = this._handle.getAsyncId();
+  state.handle = handle;
```

Comment:
It feels a bit cleaner if these were all declared directly on the `state` object. Meaning:

```js
const state = {
  handle,
  receiving: false,
  ...etc,
};
```